src/cache/cloud_fraction.jl

```diff
@@ -0,0 +1,159 @@
+import StaticArrays as SA
+import ClimaCore.RecursiveApply: rzero, ⊞, ⊠
+
+# TODO: write a test with scalars that are linear with z
+"""
+    Diagnose horizontal covariances based on vertical gradients
+    (i.e. taking turbulence production as the only term)
+"""
+function covariance_from_grad(coeff, ᶜmixing_length, ᶜ∇Φ, ᶜ∇Ψ)
+    return 2 * coeff * ᶜmixing_length^2 * dot(ᶜ∇Φ, ᶜ∇Ψ)
+end
+
+"""
+   Compute the grid scale cloud fraction based on sub-grid scale properties
+"""
+function set_cloud_fraction!(Y, p, ::DryModel)
+    FT = eltype(Y)
+    (; ᶜcloud_fraction) = p.precomputed
+    @. ᶜcloud_fraction = FT(0)
```

Comment:
```suggestion
    @. p.precomputed.ᶜcloud_fraction = 0
```
I don't think type casting is needed for `Int`s, since it should fall back on Julia's type promotion. Also, `ᶜcloud_fraction` already has an element type, which cannot change.